src/main/java/com/iexec/core/task/TaskController.java

```diff
@@ -34,22 +42,32 @@
 import static org.springframework.http.ResponseEntity.notFound;
 import static org.springframework.http.ResponseEntity.ok;
 
+@Slf4j
 @RestController
 public class TaskController {
 
-    private final TaskService taskService;
+    private final EIP712ChallengeService challengeService;
+    private final IexecHubService iexecHubService;
     private final ReplicatesService replicatesService;
     private final TaskLogsService taskLogsService;
+    private final TaskService taskService;
 
-    public TaskController(TaskService taskService,
+    public TaskController(EIP712ChallengeService challengeService,
```

Comment:
In a next PR we could rename this to 
```suggestion
    public TaskController(EIP712ChallengeService eip712ChallengeService,
```
To not be confused with the other challenge service.

Reply:
no problem, done in  5247dbf